src/durmodbin/nnet3-durmodel-rescore-lattice.cc

```diff
@@ -0,0 +1,130 @@
+// durmodbin/nnet3-durmodel-rescore-lattice.cc
+
+// Copyright 2015 Hossein Hadian
+
+// See ../../COPYING for clarification regarding multiple authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+#include "base/kaldi-common.h"
+#include "util/common-utils.h"
+#include "fstext/fstext-lib.h"
+#include "lat/kaldi-lattice.h"
+#include "lat/lattice-functions.h"
+#include "hmm/transition-model.h"
+#include "durmod/kaldi-durmod.h"
+
+int main(int argc, char *argv[]) {
+  try {
+    using namespace kaldi;
+    typedef kaldi::int32 int32;
+    using fst::SymbolTable;
+    using fst::VectorFst;
+    using kaldi::CompactLatticeArc;
+
+    const char *usage =
+      "Rescore a lattice using the scores from a phone duration model.\n"
+      "Usage: nnet3-durmodel-rescore-lattice [options] <nnet3-dur-model> "
+      "<trans-model> <lattice-rspecifier> <lattice-wspecifier>\n"
+      "e.g.: \n"
+      "nnet3-durmodel-rescore-lattice 20.mdl final.mdl "
+      "ark:lat.1 ark:rescored_lat.1\n";
+
+    BaseFloat lm_scale = 1.0;
+    ParseOptions po(usage);
+    po.Register("lm-scale", &lm_scale, "Scaling factor for language model "
+                "costs");
```

Comment:
Please don't call this lm-scale, because it's not really the lm-scale, it's the duration-model score's scaling factor.  You could call it duration-model-scale perhaps.  The same goes for the scripts.